packages/marshal/src/encodeToCapData.js

```diff
@@ -435,12 +433,10 @@ export const makeDecodeFromCapData = ({
           // Don't harden since we're not done mutating it
           const result = { [QCLASS]: decodeFromCapData(original) };
           if (hasOwnPropertyOf(jsonEncoded, 'rest')) {
-            assert(
-              typeof rest === 'object' &&
-                rest !== null &&
-                ownKeys(rest).length >= 1,
-              X`Rest encoding must be a non-empty object: ${rest}`,
-            );
+            (typeof rest === 'object' &&
+              rest !== null &&
+              ownKeys(rest).length >= 1) ||
```

Comment:
up above you had `if (rest === null) {` instead of `||`. That made it a little easier to read.

Consider,
```
const isNonEmptyObject = (typeof rest === 'object' &&
…
isNonEmptyObject || Fail`Rest encoding must be a non-empty object: ${rest}`;
```

Reply:
I like it. Done.